frc46_factory_token/token_impl/src/lib.rs

```diff
@@ -106,25 +110,22 @@ pub struct ConstructorParams {
     pub minter: Address,
 }
 
-pub fn construct_token(params: ConstructorParams) -> Result<u32, RuntimeError> {
-    let runtime = ActorRuntime::<FvmSyscalls, Blockstore>::new_fvm_runtime();
+pub fn construct_token<S: Syscalls, BS: Blockstore>(
+    runtime: ActorRuntime<S, BS>,
+    params: ConstructorParams,
+) -> Result<u32, RuntimeError> {
     let minter = runtime.resolve_id(&params.minter)?;
     let token =
-        FactoryToken::new(&runtime, params.name, params.symbol, params.granularity, Some(minter));
+        FactoryToken::new(runtime, params.name, params.symbol, params.granularity, Some(minter));
 
     let cid = token.save()?;
-    fvm_sdk::sself::set_root(&cid)?;
+    fvm_sdk::sself::set_root(&cid).map_err(|_| NoStateError)?;
```

Comment:
can this be swapped with a function from `ActorRuntime` since we have access to one here